internal/frontend/selfservice.go

```diff
@@ -78,37 +133,56 @@ func useSelfServiceForm(n core.Nexus) HandlerStep {
 					Value: userEMailAddressSnippet.Render(user),
 				},
 				h.SelectFieldSpec{
-					Name:     "memberships",
+					Name:     Memberships,
 					Label:    "Group memberships",
 					Options:  memberships,
 					ReadOnly: true,
 				},
 				h.MultilineInputFieldSpec{
-					Name:  "ssh_public_keys",
+					Name:  sshPublicKeys,
 					Label: "SSH public key(s)",
 				},
 				h.FieldSet{
-					Name:       "change_password",
+					Name:       changePassword,
 					Label:      "Change password",
 					IsFoldable: true,
 					Fields: []h.FormField{
 						h.InputFieldSpec{
 							InputType: "password",
-							Name:      "old_password",
+							Name:      oldPassword,
 							Label:     "Old password",
 						},
 						h.InputFieldSpec{
 							InputType: "password",
-							Name:      "new_password",
+							Name:      newPassword,
 							Label:     "New password",
 						},
 						h.InputFieldSpec{
 							InputType: "password",
-							Name:      "repeat_password",
+							Name:      repeatPassword,
 							Label:     "Repeat password",
 						},
 					},
 				},
+				h.FieldSet{
+					Name:       configureTotp,
+					Label:      "Configure TOTP",
+					IsFoldable: true,
+					Fields: []h.FormField{
+						// TODO: change to QRCode saved into qrCode
```

Comment:
I would need a bit help on how to display the qrcode as a png

Reply:
Since the images are going to be rather compact, we can probably just encode them into data URLs: https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/Data_URLs

The result would look like `<img alt="QR code for TOTP enrollment" src="data:image/png;base64,$DATA">`.